nixos/modules/system/boot/systemd.nix

```diff
@@ -628,7 +635,13 @@ in
 
     environment.systemPackages = [ systemd ];
 
-    environment.etc = {
+    environment.etc = let
+      generators = (fold (p: attrs: let path = "${p}/lib/systemd/system-generators";
+                                    in attrs // (mapAttrs' (name: _: nameValuePair name "${path}/${toString name}")
+                                                           (builtins.readDir "${path}")))
```

Comment:
Doesn't this cause the package to be built at evaluation time? That's not allowed. Probably you should have a derivation to generate an aggregated `lib/systemd/system-generators` directory like this:
```
runCommand "system-generators"
  { packages = cfg.generator-packages; }
  ''
    mkdir -p $out/lib/systemd/system-generator
    for i in $packages; do
      ln -s $i/lib/systemd/system-generator/* $out/lib/systemd/system-generator
    done
  '';
```
I.e. similar to how we handle units.